test/tstUnionFind.cpp

```diff
@@ -0,0 +1,124 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+#include "ArborX_EnableDeviceTypes.hpp" // ARBORX_DEVICE_TYPES
+#include "ArborX_EnableViewComparison.hpp"
+#include <ArborX_DetailsUnionFind.hpp>
+#include <ArborX_DetailsUtils.hpp> // iota
+
+#include "BoostTest_CUDA_clang_workarounds.hpp"
+#include <boost/test/unit_test.hpp>
+
+BOOST_AUTO_TEST_SUITE(UnionFind)
+
+namespace
+{
+
+template <typename ExecutionSpace, typename UnionFind>
+Kokkos::View<int *, Kokkos::HostSpace>
+build_representatives(ExecutionSpace const &space, UnionFind union_find)
```

Comment:
Why "build" rather than "find"?

Reply:
Because I consider "find" to build a call to the union-find structure. Here, we construct an additional view outside of the union-find structure containing representatives. So I think "build" is appropriate.

Reply:
I am not convinced but not blocking